tasm/tiles/include/LayoutDatabase.h

```diff
@@ -0,0 +1,62 @@
+#ifndef TASM_LAYOUTDATABASE_H
+#define TASM_LAYOUTDATABASE_H
+
+#include <experimental/filesystem>
+#include "sqlite3.h"
+#include "TileLayout.h"
+#include "Video.h"
+#include "Files.h"
+#include "EnvironmentConfiguration.h"
+#include <gtest/gtest.h>
+
+namespace tasm {
+
+class LayoutDbConfiguration {
+public:
+    static const std::experimental::filesystem::path &LayoutDbPath() {
+        return EnvironmentConfiguration::instance().defaultLayoutDatabasePath();
+    }
+};
+
+class LayoutDatabase {
+public:
+    static std::shared_ptr<LayoutDatabase> &instance() {
```

Comment:
Because this is a pointer, we don't need to return a reference. However, if we want to enforce that callers of this can't change it, we can add some `const` annotations.
Option 1: return a const reference `const LayoutDatabase&`
Option 2: return a shared_ptr to a constant object: `std::shared_ptr<const LayoutDatabase>`
Option 3: return a constant shared_ptr to a non-constant object: `const std::shared_ptr<LayoutDatabase>`
Option 4: return a constant shared_ptr to a constant object: `const std::shared_ptr<const LayoutDatabase>`
Option 4 means that a caller can't re-assign the pointer, and they also can't modify the pointed-to object. 

I haven't been good about putting `const` in or around pointers in the rest of the project, so I don't feel super strongly about having to add them here.